src/Stack/Docker.hs

```diff
@@ -229,10 +229,12 @@ runContainerAndExit = do
      when (isNothing mpath) $ do
        logWarn "The Docker image does not set the PATH env var"
        logWarn "This will likely fail, see https://github.com/commercialhaskell/stack/issues/2742"
-     newPathEnv <- either throwM return $ augmentPath
-                      [ hostBinDir
-                      , toFilePath (sandboxHomeDir </> relDirDotLocal </> relDirBin)]
-                      mpath
+     newPathEnv <- either throwM return $
+                   bool id (T.replace ";" ":") osIsWindows <$>
```

Comment:
Personally I find this usage of `bool` ambiguous. I'd rather see a helper function defined above using explicit if-then-else or pattern matching.

Reply:
I will change that

Reply:
Replaced with helper function :)